src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/SpeechToTextImplementation.macos.cs

```diff
@@ -80,17 +84,26 @@ async Task<string> InternalListenAsync(CultureInfo culture, IProgress<string>? r
 					{
 						var s = result.BestTranscription.Segments[i].Substring;
 						currentIndex++;
-						recognitionResult?.Report(s);
+						recognitionProgress?.Report(s);
+						OnRecognitionResultUpdated(s);
 					}
 				}
 			}
 		});
+		return Task.CompletedTask;
```

Comment:
Why not return the `getRecognitionTaskCompletionSource.Task`?

Reply:
this method doesn't use getRecognitionTaskCompletionSource. if you call this method, getRecognitionTaskCompletionSource is null.

Reply:
Probably I'm not following the flow, so how it's null if it's used by `recognitionTask` at line 69? And also looks like the only place that it gets instantiated is on `InternalListenAsync` method.

Reply:
it is marked as nullable there.
we have 2 methods: IntenalListenAsync and InternalStartListenAsync.
- InternalListenAsync is waiting for the result, so it needs that getRecognitionTaskCompletionSource. (and it initialize it, then calls InternalStartListenAsync, In that case InternalStartListenAsync has defined getRecognitionTaskCompletionSource)
- InternalStartListenAsync is called directly from new method StartListenAsync, so this getRecognitionTaskCompletionSource is not initialized in that case and just ignored

Reply:
Any actions here? This is the final open conversation on this PR before we can merge it. 

Given the [security vulnerability that we patched on SkiaSharp today](https://github.com/CommunityToolkit/Maui/issues/1424), I'd like to publish a new release of `CommunityToolkit.Maui` as quickly as possible so that our users are subsequently patched!

Reply:
For me it's good to go